cylc/flow/scheduler.py

```diff
@@ -1525,7 +1524,7 @@ async def update_data_structure(self):
         updated_nodes = set(updated_tasks).union(
             self.pool.get_pool_change_tasks())
         if (
-                has_updated or
+                updated_nodes or
```

Comment:
The datastore wasn't updating on some changes (noticed via tests that use `cylc dump`, which now goes via GraphQL to the datastore).